### PR TITLE
11007 isPrime does not work for large numbers

### LINKED ISSUE #11007: isPrime does not work for large numbers
**Bug description**
`(105850338e131072+1) isPrime` is causing an error `Error: Cannot truncate this number` after a while.

### LINKED ISSUE #11046: Integer>>isPrime isBroken
Some idiot (yes it was me) has broken isPrime, it fails for specific squares  Obvious bug in hindsight.  `121 isPrime = true` obviously brain still fluffed from Covid.  PR incoming.

### PR DESCRIPTION
**Pull request by johnhwoods**
`LargeInteger>>isPrime` overrides `Integer>>isPrime` with `^self isProbablyPrime` - a 25 iteration Miller/Rabin test.

However, because the fractional indices in the algorithm were expressed as floats, this stopped `isPrime` being useful above `Float fmax` (approx 1.8e308).  However, there's no performance reason for such a ceiling (and, indeed, many of the LargeIntegers we want to test for probable primality are cryptographic numbers in excess of 2^1024 and therefore larger than this).   The new method will return true, in a minute or so, on average 2020 PC, for the 23rd Mersenne Prime, 2^11213-1, a 3376 digit number.

### CODE REVIEW

File: src/Math-Operations-Extensions/Integer.extension.st [hunk 1]
```diff
@@ -146,29 +146,18 @@ Integer >> isPrime [
 Integer >> isProbablyPrime [
 	"See isProbablyPrimeWithK:andQ:randomIndex: for the algoritm description."
```
Comment (astares) [inline src/Math-Operations-Extensions/Integer.extension.st]: can you change typo: algorithm

Comment (johnhwoods) [reply src/Math-Operations-Extensions/Integer.extension.st]: Hah, missed that.  In my defence that typo is in the original method :-)

**Comment by MarcusDenker** [comment]
failig test not related:

- [CI] Failing randomly: Spec related tests with DNU SmallFloat64 >> #nextCell: #10522

**Comment by Ducasse** [comment]
Does it fix #11046

**Comment by Ducasse** [comment]
I'm confused what is the difference with PR about isPrime that got integrated

**Comment by johnhwoods** [comment]
Hi, there were three separate issues which I idiotically turned into four.

1) isPrime didn't work for large numbers (where it redirects to isProbablyPrime which used a method which contained floats instead of fractions for its random indices in a Miller/Rabin method.  I changed the floats to fracs and it worked)  This was #11007, opened by Pavel.

2) I noticed when doing it that the SmallInteger isPrime had a 2x inefficiency so I raised #11018, and fixed this.

3) I also noticed primeFactors could be improved, so I raised #11045.  Whilst doing this I realised that

4) Unfortunately when I did (2) I introduced a stupid bug.  So I raised #11046.

Have I earned a ban yet?  :-D

**Review by MarcusDenker — APPROVED** [review]

**Pull request merged**